package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn

```diff
@@ -8,16 +8,19 @@ local uci = require('simple-uci').cursor()
 local unistd = require 'posix.unistd'
 
 
-uci:section('network', 'interface', 'mesh_vpn', {
-	ifname = 'mesh-vpn',
-	proto = 'gluon_mesh',
-	transitive = true,
-	fixed_mtu = true,
-	macaddr = util.generate_mac(7),
-	mtu = site.mesh_vpn.mtu(),
-})
-
-uci:save('network')
+if not unistd.access('/lib/gluon/mesh-vpn/wireguard') then
+	-- wireguard brings its own mechanism for creating interfaces as it
+	-- requires one interface per peer.
+	uci:section('network', 'interface', 'mesh_vpn', {
+		ifname = 'mesh-vpn',
+		proto = 'gluon_mesh',
+		transitive = true,
+		fixed_mtu = true,
+		macaddr = util.generate_mac(7),
+		mtu = site.mesh_vpn.mtu(),
+	})
+	uci:save('network')
+end
```

Comment:
While wireguard doesn't use the section, there is no harm in creating it either to avoid more conditional UCI config.

Reply:
What does that mean - would you want me to change anything or is this rather a comment?t